### DISCUSSION TITLE
Multi-currency price

### DISCUSSION THREAD

**Discussion by armezit** [lunarphp/lunar#763] [Feature Requests]
I'm going to develop an add-on to auto update a multi-currency store prices.

I'm thinking of implementing two model observers for `Currency` and `ProductVariant`.
so, I'd be able to recalculate prices whenever:

- Currency exchange rate changes
- Variant price changes

Do you think this is the proper way?

Also I'd like to know if there are any plans to support these kinds of multi-currency features in the Lunar? either in the core or official add-ons?

**Comment by glennjacobs** [comment]
I would imagine you'd want to observe the Price model as well as Currency model. The obvious factor is that if a Currency is updated, then it's got to update potentially 1,000's of prices. So it would want to be run in the queue as a background job.

I think this is something we would like in the core and I believe we already have an issue open for it. See https://github.com/lunarphp/lunar/issues/383

**Reply by armezit** [reply]
Good point @glennjacobs. Price model instead of ProductVariant.

I saw #383 a few months ago, but it seemed to me that it is not a priority.

Do you recommend to implement this in core and maybe a PR ? or as separate add-on?

**Reply by glennjacobs** [reply]
I would say the core as a PR, as otherwise, the currency conversion values are pretty redundant without it. 

We need to consider how we will show admin users in the hub that the prices are still updating. It almost needs a background jobs section in the hub for this sort of thing, but we don't have that yet.

**Reply by armezit** [reply]
@glennjacobs I've created a basic PR for price conversion: https://github.com/lunarphp/lunar/pull/771
If you got time, maybe you can take a look.
